relation_engine_server/test/test_json_validation.py

```diff
@@ -85,44 +95,243 @@ def test_non_validation_validator_errors(self):
                 validate_at='/properties/params')
         self.assertEqual(output, {'name': 'name', 'distance': 3})
 
-    def test_json_validation_defaults(self):
+    def test_json_validation(self):
+
+        test_list = [
+            self.add_defaults,
+            self.pattern_validation,
+            self.uri_validation,
+            self.date_format_validation,
+        ]
+
+        for test_schema in test_schema_list:
+            schema_file_arg = schema_arg = test_schema[1]
+
+            if test_schema[0] == 'schema':
+                schema_file_arg = None
+            else:
+                schema_arg = None
+
+            for test in test_list:
+                test(schema_arg, schema_file_arg)
```

Comment:
Run all the tests in `test_list` using a python data structure for the schema or by retrieving the schema from a JSON or YAML file.